### PR TITLE
Remove net451 as a cross-compile target

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: src/Microsoft.AspNetCore.TestHost/ResponseStream.cs [hunk 1]
```diff
@@ -266,24 +253,7 @@ public override void Write(byte[] buffer, int offset, int count)
                 _writeLock.Release();
             }
         }
-#if NET451
-        public override IAsyncResult BeginWrite(byte[] buffer, int offset, int count, AsyncCallback callback, object state)
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.TestHost/ResponseStream.cs]: Do we care about this in TestHost?

Comment (cesarbs) [inline src/Microsoft.AspNetCore.TestHost/ResponseStream.cs]: We ran into deadlocks in Kestrel when not implementing these methods on full framework. Not sure if that applies here though.

Comment (Tratcher) [inline src/Microsoft.AspNetCore.TestHost/ResponseStream.cs]: Yes, keep it.

File: test/Microsoft.AspNetCore.TestHost.Tests/Microsoft.AspNetCore.TestHost.Tests.csproj [hunk 2]
```diff
@@ -3,14 +3,18 @@
...
+    <TargetFrameworks>netcoreapp2.0;net46</TargetFrameworks>
     <TargetFrameworks Condition=" '$(OS)' != 'Windows_NT' ">netcoreapp2.0</TargetFrameworks>
   </PropertyGroup>
 
   <ItemGroup>
     <ProjectReference Include="..\..\src\Microsoft.AspNetCore.TestHost\Microsoft.AspNetCore.TestHost.csproj" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(TargetFramework)'=='net46'">
+    <PackageReference Include="System.Net.WebSockets" Version="$(CoreFxVersion)" />
```
Comment (davidfowl) [inline test/Microsoft.AspNetCore.TestHost.Tests/Microsoft.AspNetCore.TestHost.Tests.csproj]: ?

Comment (pranavkm) [inline test/Microsoft.AspNetCore.TestHost.Tests/Microsoft.AspNetCore.TestHost.Tests.csproj]: 👍  Wasn't needed after all.

**Comment by pranavkm** [comment]
🆙 📅

File: src/Microsoft.AspNetCore.Hosting/Microsoft.AspNetCore.Hosting.csproj [hunk 3]
```diff
@@ -26,16 +26,14 @@
...
+    <PackageReference Include="Microsoft.Extensions.StackTrace.Sources" Version="$(AspNetCoreVersion)" PrivateAssets="All" />
+    <PackageReference Include="Microsoft.Extensions.TypeNameHelper.Sources" Version="$(AspNetCoreVersion)" PrivateAssets="All" />
     <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="$(CoreFxVersion)" />
-    <PackageReference Include="System.Reflection.Metadata" Version="$(SystemReflectionMetadataVersion)" />
-  </ItemGroup>
-
-  <ItemGroup Condition="'$(TargetFramework)' != 'net451'">
     <PackageReference Include="System.Diagnostics.StackTrace" Version="$(CoreFxVersion)" />
+    <PackageReference Include="System.Reflection.Metadata" Version="$(SystemReflectionMetadataVersion)" />
+    <PackageReference Include="System.Reflection.TypeExtensions" Version="$(CoreFxVersion)" />
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Hosting/Microsoft.AspNetCore.Hosting.csproj]: This is the bug you were talking about?

**Pull request merged**